AK/MemoryStream.cpp

```diff
@@ -93,7 +93,9 @@ ErrorOr<size_t> FixedMemoryStream::seek(i64 offset, SeekMode seek_mode)
 
 ErrorOr<size_t> FixedMemoryStream::write_some(ReadonlyBytes bytes)
 {
-    VERIFY(m_writing_enabled);
+    // Match the POSIX error code for non-writable mapped memory.
+    if (!m_writing_enabled)
+        return Error::from_errno(EACCES);
```

Comment:
As discussed on Discord:

```suggestion
    // MemoryStream isn't based on file-descriptors, but since most other
    // Stream implementations are, the interface specifies EBADF as the
    // "we don't support this particular operation" error code.
    if (!m_writing_enabled)
        return Error::from_errno(EBADF);
```

The commit message needs to be rewritten accordingly.